### ISSUE TITLE
Ability to tune down/up monster drops

### ISSUE THREAD

**Issue by Warlander** [CleverRaven/Cataclysm-DDA#28919]
**Is your feature request related to a problem? Please describe.**  
Currently, you can only modify item spawn rates for items spawning in buildings, and there is no similar setting for monster drops. When playing with lowered item spawn rate settings, amount of loot dropped by monsters is unbalanced.

**Describe the solution you'd like**  
Additional setting for tuning monster drops up or down, similar to "Item spawn scaling factor".

**Describe alternatives you've considered**  
Having "Item spawn scaling factor" influence both building and monster drops. That's one possible solution, but gives less freedom for players who would like to have separate ratios for them.

**Comment by DSeyka** [comment]
It definitely feels like zombie drops are much more generous than they used to be, especially in regards to batteries and electronic stuff in general.

**Comment by Warlander** [comment]
Yes, loot from zombies seem to be much higher indeed - currently single day raid to the mall is enough to get you thousands on batteries, enough to last for the entire game until you power everything with UPS or bionic power.

**Comment by Night-Pryanik** [comment]
That's the problem of our universal batteries system.

**Comment by xrandomuserx** [comment]
Also agree. Have played only 0.4 item spawn for a long time now. As of recent builds, much more loot from monsters, especially phone/smartphones. If you manage to find a crashed copter, you can easily* kill some zombie soldiers and get OP weapons (damaged).

The phones I understand, but yes much more loot from monsters, very unbalanced.

**Comment by Night-Pryanik** [comment]
> much more loot from monsters, very unbalanced.

We could tune it down. Please describe what loot in particular you encounter too much and want it to be lower.

**Comment by Warlander** [comment]
> > much more loot from monsters, very unbalanced.
> 
> We could tune it down. Please describe what loot in particular you encounter too much and want it to be lower.

Ideal solution is just having option to regulate loot from monsters, but if that's not possible or easy - there are way too many electronics, and they often drop with lots of batteries in them (especially 120/120 cellphones are common). Permanent markers, lighters, refillable lighters and matchboxes drop in overabundance, too. Ammo drops a lot as well, but that's IMHO a good thing with how hard it is to craft - gun drops could be tuned down a bit, through.

**Comment by DSeyka** [comment]
Another thing I've noticed, is that battery count is only sometimes randomized. Every single talking doll I've found has 20/100 batteries in it, for example.

**Comment by Night-Pryanik** [comment]
> there are way too many electronics, and they often drop with lots of batteries in them (especially 120/120 cellphones are common)

Don't you see people carry lots of electronic devices IRL today?

> lighters, refillable lighters and matchboxes drop in overabundance, too.

Don't you see lots of smokers on the streets IRL today?

> Another thing I've noticed, is that battery count is only randomized for particular items, and never for others. Every single talking doll I've found has 20 out of 100 batteries in it, for example.

That's just because charges in some items are randomized, and in others they ain't.

**Comment by DSeyka** [comment]
> That's just because charges in some items are randomized, and in others they ain't.
Is this intended though? Seems like an oversight to me.

**Comment by Night-Pryanik** [comment]
> Is this intended though? Seems like an oversight to me.

Of course it's not intended, nor it's an oversight. It's just "I'll give it 20 charges, and I'm done with it".

**Comment by DSeyka** [comment]
Fair enough.

**Comment by Warlander** [comment]
> Don't you see people carry lots of electronic devices IRL today?

Yes, I doubt you can just extract batteries from them like in CDDA, through. Maybe instead of working devices you could often get random part(s) you would get from their disassembly?

> Don't you see lots of smokers on the streets IRL today?

Can't argue with that, amount of working lighters and packs you get is insane, through. Maybe at least getting rid of packs to reduce inventory micromanagement could be a good idea? (the same way we don't have antibiotics etc. in packs even through they are IRL).

**Comment by xrandomuserx** [comment]
Speaking of cigarettes. I should be able to reload / part reload my pack of cigarettes with the other loose cigarettes in my inventory.

**Comment by xrandomuserx** [comment]
https://www.reddit.com/r/cataclysmdda/comments/ao8tc8/zombies_drop_too_much_loot/
Current world, I have all sorts of bracelets and rings and necklaces too...
Not sure (at all) how code is turned into a working game. But maybe some simple algebra?
Where monster drop chance x loot drop scale = actual drop chance

[
  {
    "id": "mon_zombie_soldier_death_drops",
    "type": "item_group",
    "subtype": "collection",
    "magazine": 100,
    "ammo": 60,
    "entries": [
      { "distribution": [
        { "item": "LAW_Packed", "prob": 20X },
        { "item": "e_tool", "prob": 100X },
        { "item": "mask_gas", "prob": 100X, "damage-min": 1, "damage-max": 4 },
        { "item": "two_way_radio", "prob": 50X },
{ "item": "c4", "prob": 10X }

Where X = Loot drop scale (0.01 to 2 or 3)


TO NOTE, modern phones are not user serviceable, and the batteries cannot be unloaded. Consider phones to have lithium ion batteries instead of standard batteries. You can always modify a phone with battery mod or similar - not that this is expected, i understand battery is another (large) topic.

**Comment by kevingranade** [comment]
> Ideal solution is just having option to regulate loot from monsters

This is not ideal at all, in particular related to the comments:
> loot from zombies seem to be much higher
> much more loot from monsters

this impacts many players that want to play low-loot games, we want to fix it for everyone, not just people willing to tune their game.

**Comment by Warlander** [comment]
Yes, more loot present in current versions impacts players who want to play low-loot games. Is there any other way to limit loot drops for them other than possibility to tune the game like in case of item spawn rate, through?

I think there is need for some balance changes in loot for everyone, but it wouldn't hurt to have an option to globally reduce them for extra challenge or slower-paced game.

**Comment by xrandomuserx** [comment]
[url=https://ibb.co/F0M31vN][img]https://i.ibb.co/F0M31vN/Grappler.jpg[/img][/url]
Grappler Zombie with Abundant Loot

[url=https://ibb.co/72tBbB8][img]https://i.ibb.co/72tBbB8/Bloated.jpg[/img][/url]
Bloated Zombie with Abundant Loot - Or a sling pack full of loot?

Perhaps with loot.
Spawning one item - 70% -> spawning another item 50% -> spawning another item 30% ->spawning another item 20%

Chance of spawning four items 2%

Or however smart coders write code so monsters are less likely to drop multiple loot items...
At this stage seems to be a chance to drop any item, or similar....

**Comment by kevingranade** [comment]
> it wouldn't hurt to have an option to globally reduce them for extra challenge or slower-paced game.

It would, because the more different options we have, the less able we are to test them all. More options means more bugs and less balance. Extending an existing option to do this would work fine though.

**Comment by Night-Pryanik** [comment]
> Grappler Zombie with Abundant Loot

A set of filthy clothes. Inhaler, permanent marker in one pocket, flyer, rolling paper and plastic bags in the other. Game watch on the wrist. 

> Bloated Zombie with Abundant Loot. Or it carried a sling pack full of loot?

Yes, it is.

What's so "abundant" with their loot? Is there something _really_ out of sense?

**Comment by xrandomuserx** [comment]
Well these are only a few examples. I have played, for a very long time with low item spawns. Sometime recently it has been changed, and now zombies drop much higher loot. With item spawn set at one, it seems like I am the only survivor, shops fully stocked, so I turn it down. Now every few zombies I am getting given free loot and ammo. Now, I can overpower basic enemies in the first few days, except Skeletal Juggernaut, but I am sure my C-4 or Granade will kill that thing..

I could spend all day giving examples, but those are the two that appeared easily for me. The plastic bag had rotten food, but that went away after reload (to do the screenshots).

If you can easily change loot drop rates, that would be great. Perhaps lower probability to drop multiple items, or probability of dropping items related to current item spawn setting.

Would be nice.

**Comment by Maddremor** [comment]
RE: Batteries and electronics. 
Given that the cataclysm started several (5?) days before the game start, devices that are not explicitly turned off might have run out of power, especially if they have been recieving alert messages from the government during the crisis. Additionally, zombies are not inclined to protect their devices from water and physical damage, so only some of them would be functional.

**Comment by Night-Pryanik** [comment]
> Given that the cataclysm started several (5?) days before the game start, devices that are not explicitly turned off might have run out of power

For ordinary zombies I made 90% chance to drop fully discharged phones, and only 10% of phones will spawn with 0..15 charges in them. Is this too much, you think?

**Comment by Maddremor** [comment]
I would probably assume that you can't extract batteries from a broken phone, and then apply the 0.1 chance to the intact ones.

**Comment by Inglonias** [comment]
The battery system in Cataclysm is a relic of an earlier age in the game's development. It's certainly a little bit abstract these days.

**Comment by DaviBones** [comment]
I'd like to fix this by rolling against each item and getting rid of it X% of the time, where
```X = 1 - spawn rate scaling factor```.

Cons:
- In order to avoid adding a special case to the item generation engine, this solution would remove the items only after they are already created. It could be argued that this is wasting a few CPU cycles
- Will sometimes result in zombies with no pants

Pros:
- Easy to implement and maintain, since it utilizes the existing setting.
- Solves issue for those who like to play on low item spawns
- Will sometimes result in zombies with no pants

...Any thoughts/objections before I prepare a PR?

**Comment by Night-Pryanik** [comment]
We need to be _absolutely_ sure that we don't get no "natural" monster drops (let's say, some robot not dropping its integral circuits) on its death after a while with your solution.

**Comment by DaviBones** [comment]
> We need to be _absolutely_ sure that we don't get no "natural" monster drops (let's say, some robot not dropping its integral circuits) on its death after a while with your solution.

You mean like a `skitterbot` not dropping `broken skitterbot`? My solution doesn't touch `revert_to_itype` so that shouldn't be a problem.

Or did you mean something else?

**Comment by Night-Pryanik** [comment]
I'm not at my PC now, so I can't tell for sure, but IIRC manhacks, for example, are dropping spikes on death.

**Comment by DaviBones** [comment]
Well, manhacks always drop a single item (besides the broken manhack itself), chosen from a pretty large pool of options, including heavy batteries, plutonium cells, kevlar plates, etc. A spike will spawn exactly 8% of the time. Incidentally, none of these are required in the crafting recipe of the inactive manhack, and therefore make no sense from a realism standpoint (or a balance standpoint for that matter, since you could theoretically farm plutonium by repeatedly crafting, destroying, and disassembling manhacks).

All the realism-driven drops are already covered by disassembly of the broken robot itself, which again, my code doesn't touch.
